packages/adapter-vercel/index.js

```diff
@@ -141,11 +156,44 @@ const plugin = function ({ external = [], edge, split } = {}) {
 
 			builder.log.minor('Writing routes...');
 
-			write(`${dir}/config.json`, JSON.stringify(config, null, '  '));
+			write(`${dir}/config.json`, JSON.stringify(static_config, null, '  '));
 		}
 	};
 };
 
+/**
+ * @param {import('.').Config | undefined} config_a
+ * @param {import('.').Config | undefined} config_b
+ */
+function can_group(config_a, config_b) {
+	if (config_a === config_b) return true;
+	if (!config_a || !config_b) return false;
+
+	if (config_a.runtime !== config_b.runtime) return false;
+	if (config_a.maxDuration !== config_b.maxDuration) return false;
+	if (config_a.memory !== config_b.memory) return false;
```

Comment:
I kinda think `memory` should mean 'make this route its own function'. The only reason you'd set it is if a particular route has unusually high memory requirements — if you have three particularly intensive API routes that need 1024Mb each, then merging them all into a single function will defeat the object.

Reply:
Lambda allocates CPU in proportion to the amount of `memory` configured, so it can make sense to increase/decrease `memory` for a `group` without the need of splitting, to optimise startups times for instance. Also, decreasing `memory` for a particular non critical part of the app can be useful to limit GB-Hrs usage.

Reply:
I think that the `group` option may live nicely with the '_auto_' bundling strategy by adding another case in `can_group()`.

Reply:
Thinking further, I suppose combining those intensive routes into one function _isn't_ generally a problem in the lambda world, because one lambda will only serve one request at a time. So maybe we can combine them after all?

Then again if the memory is used while the function is warm but not active... gah

Reply:
I do still prefer the `group` over `split: true` at a route level as it covers more uses cases. Applying `split: true` at a layout level will split all subsequent routes, the `group` option allows you to do the same by defining a unique group on each route or bundling all subsequent routes by defining it at a layout level or by bundling routes that are not even on the same route branch. I find it more versatile.

Reply:
I may be overthinking about this.. Other metaframeworks don't offer that much versatility and I don't see that much complains.

Reply:
Ultimately the way functions are bundled together is something that you ideally wouldn't have to think about — the system should be smart enough to do the right thing (e.g. with `split: false`, automatically splitting _anyway_ when a function exceeds a size limit), which is why I lean towards an escape hatch over 'another thing the developer needs to think about'